Month1/Week2/Steam/Steam/CategoryContainer.cpp

```diff
@@ -0,0 +1,52 @@
+#include "CategoryContainer.h"
+
+//-------------------------------------------------------------------------------------------
+FCategoryContainer::FCategoryContainer()
+{
+	CategoryCount = 0;
+}
+
+//-------------------------------------------------------------------------------------------
+int FCategoryContainer::GetCategoryCount()
+{
+	return CategoryCount;
+}
+
+//-------------------------------------------------------------------------------------------
+FCategory FCategoryContainer::GetCategory(int index) const
+{
+	return Categories[index];
+}
+
+//-------------------------------------------------------------------------------------------
+bool FCategoryContainer::AddCategory(FCategory pCategory)
+{
+	if (CategoryCount < MaxCategoryCount)
+	{
+		Categories[CategoryCount] = pCategory;
+		CategoryCount++;
+		return true;
+	}
+	return false;
+}
+
+//-------------------------------------------------------------------------------------------
+bool FCategoryContainer::DeleteCategory(int Index)
+{
+	if (Index < CategoryCount)
+	{
+		//Cascade the categories into their new positions. This should end up deleting the indexed one as well.
+		for (; Index < CategoryCount && Index < MaxCategoryCount - 1; Index++)
+		{
+			Categories[Index] = Categories[Index + 1];
+		}
+		CategoryCount--;
+		return true;
+	}
+	return false;
+}
```

Comment:
I think it is best to by default introduce a new variable `i`. In this case it isn't really needed, but it would be preferred for other cases where you do want to use the original Index in another place in the function, for example.